src/compiler/checker.ts

```diff
@@ -5458,8 +5474,16 @@ module ts {
             // will be bound to non-arrow function that contain this arrow function. This results in inconsistent behavior.
             // To avoid that we will give an error to users if they use arguments objects in arrow function so that they
             // can explicitly bound arguments objects
-            if (symbol === argumentsSymbol && getContainingFunction(node).kind === SyntaxKind.ArrowFunction && languageVersion < ScriptTarget.ES6) {
-                error(node, Diagnostics.The_arguments_object_cannot_be_referenced_in_an_arrow_function_in_ES3_and_ES5_Consider_using_a_standard_function_expression);
+            let container = getContainingFunction(node);
+            if (symbol === argumentsSymbol) {
+                if (container.kind === SyntaxKind.ArrowFunction) {
+                    if (languageVersion < ScriptTarget.ES6) {
+                        error(node, Diagnostics.The_arguments_object_cannot_be_referenced_in_an_arrow_function_in_ES3_and_ES5_Consider_using_a_standard_function_expression);
+                    }
+                    else if (node.parserContextFlags & ParserContextFlags.Await) {
+                        error(node, Diagnostics.The_arguments_object_cannot_be_referenced_in_an_async_arrow_function_Consider_using_a_standard_async_function_expression);

```

Comment:
No, the "arguments" I pass to the generator are the arguments to the arrow function, captured via `(...args_1) => ...`. I need to do this so that any binding patterns are processed in the context of the generator function. As a result, `arguments` inside the body of the function would point to the `arguments` of the generator and not the lexical `arguments` that should have been captured by the arrow. We can look into changing this later, but for now we just report an error.